bokeh/plotting_helpers.py

```diff
@@ -318,6 +318,71 @@ def _tool_from_string(name):
 
         raise ValueError("unexpected tool name '%s', %s tools are %s" % (name, text, nice_join(matches)))
 
+
+def _add_tools_to_plot(plot, tools, tools_to_remove=None):
+    """ Adds tools to the plot object skipping those specified in
+    tools_to_remove.
+
+    Args:
+        plot (Plot): instance of a plot object
+        tools (list(Tool)|str): list of tool types or string listing the
```

Comment:
I use `seq[foo]` when a generic sequence is accepted. So this could be:
```
seq[Tool or str]
```
which is actually the most correct since I think its OK to mix them.